src/units/unit.cpp

```diff
@@ -1421,6 +1421,15 @@ void unit::remove_ability_by_id(const std::string& ability)
 	}
 }
 
+ /**
+ * Returns true if any of "add", "sub", "multiply" or "divide" are present and have a value
+ * other than their respective no-op value.
+ */
+static bool has_value_changing_modifier(const config& cfg)
+{
+	return  ((cfg.has_attribute("add") && cfg["add"].to_int(0) != 0) || (cfg.has_attribute("sub") && cfg["sub"].to_int(0) != 0) || (cfg.has_attribute("multiply") && cfg["multiply"].to_double(1) != 1) || (cfg.has_attribute("divide") && cfg["divide"].to_double(1) != 1));
```

Comment:
Please split this across multiple lines.

Reply:
done